src/frontend/src/optimizer/plan_node/batch_delete.rs

```diff
@@ -59,9 +62,43 @@ impl_distill_by_unit!(BatchDelete, core, "BatchDelete");
 
 impl ToDistributedBatch for BatchDelete {
     fn to_distributed(&self) -> Result<PlanRef> {
-        let new_input = RequiredDist::single()
+        if self
+            .core
+            .ctx()
+            .session_ctx()
+            .config()
+            .batch_enable_distributed_dml()
+        {
+            // Add an hash shuffle between the delete and its input.
+            let new_input = RequiredDist::PhysicalDist(Distribution::HashShard(
+                (0..self.input().schema().len()).collect(),
+            ))
```

Comment:
If I remember correctly, the `hash` used in batch exchange is different from streaming, and we have added a so-called `ConsistentHash` for the streaming hash distribution.

https://github.com/risingwavelabs/risingwave/blob/f85de375e7fba414a9f9f14b508fb0df2dc9bf13/proto/batch_plan.proto#L344-L345

Here which distribution does this generated `BatchExchange` follow?

Reply:
This one `HASH = 3`

Reply:
> If I remember correctly, the hash used in batch exchange is different from streaming, and we have added a so-called ConsistentHash for the streaming hash distribution.

use batch hash distribution here is ok because we still have the stream hash exchange before the materialize executor